core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/UpxCompressionBuildStep.java

```diff
@@ -52,7 +52,8 @@ public void compress(NativeConfig nativeConfig, NativeImageRunnerBuildItem nativ
 
         String effectiveBuilderImage = nativeConfig.builderImage().getEffectiveImage();
         Optional<File> upxPathFromSystem = getUpxFromSystem();
-        if (upxPathFromSystem.isPresent()) {
+        if (upxPathFromSystem.isPresent() && !nativeConfig.compression().containerBuild().orElse(false)
```

Comment:
```suggestion
        if (upxPathFromSystem.isPresent() && nativeConfig.compression().containerBuild().orElse(true)
```
inverting the term is not possible, as `containerBuild` is not inverted, as well, so the `!` negation is required, right?

Reply:
```suggestion
        if (upxPathFromSystem.isPresent() && !nativeConfig.compression().containerBuild()
```